SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -0,0 +1,841 @@
+
+  Inlet_SpeciesVars.resize(nMarker);
+  for (unsigned long iMarker = 0; iMarker < nMarker; iMarker++) {
+    Inlet_SpeciesVars[iMarker].resize(nVertex[iMarker], nVar);
+    for (unsigned long iVertex = 0; iVertex < nVertex[iMarker]; ++iVertex) {
+      for (unsigned short iVar = 0; iVar < nVar; iVar++) {
+        Inlet_SpeciesVars[iMarker](iVertex, iVar) = Solution_Inf[iVar];
+      }
+    }
+  }
+
+  /*--- Store the initial CFL number for all grid points. ---*/
+
+  const su2double CFL = config->GetCFL(MGLevel) * config->GetCFLRedCoeff_Species();
+  SU2_OMP_FOR_STAT(omp_chunk_size)
+  for (auto iPoint = 0u; iPoint < nPoint; iPoint++) {
+    nodes->SetLocalCFL(iPoint, CFL);
+  }
+  END_SU2_OMP_FOR
+  Min_CFL_Local = CFL;
+  Max_CFL_Local = CFL;
+  Avg_CFL_Local = CFL;
+
+  /*--- Add the solver name. ---*/
+  SolverName = "FLAMELET";
+}
+
+void CSpeciesFlameletSolver::Preprocessing(CGeometry* geometry, CSolver** solver_container, CConfig* config,
+                                           unsigned short iMesh, unsigned short iRKStep,
+                                           unsigned short RunTime_EqSystem, bool Output) {
+  unsigned long n_not_in_domain = 0;
+  unsigned long global_table_misses = 0;
+
+  const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+
+  CVariable* flowNodes = solver_container[FLOW_SOL]->GetNodes();
+
+  for (auto i_point = 0u; i_point < nPoint; i_point++) {
+    CFluidModel* fluid_model_local = solver_container[FLOW_SOL]->GetFluidModel();
+    su2double* scalars = nodes->GetSolution(i_point);
+
+    /*--- Compute scalar source terms ---*/
+    unsigned long exit_code = fluid_model_local->SetScalarSources(scalars);
+    unsigned short inside = exit_code;
+    nodes->SetInsideTable(i_point, inside);
+    n_not_in_domain += exit_code;
+
+    /*--- Get lookup scalars ---*/
+    fluid_model_local->SetScalarLookups(scalars);
+    for (auto i_lookup = 0u; i_lookup<config->GetNLookups(); i_lookup++) {
+      nodes->SetLookupScalar(i_point, fluid_model_local->GetScalarLookups(i_lookup), i_lookup);
+    }
+
+    for (auto i_scalar = 0u; i_scalar < nVar; i_scalar++)
+      nodes->SetScalarSource(i_point, i_scalar, fluid_model_local->GetScalarSources(i_scalar));
+
+    su2double T = flowNodes->GetTemperature(i_point);
+    fluid_model_local->SetTDState_T(T,scalars);
+    /*--- set the diffusivity in the fluid model to the diffusivity obtained from the lookup table ---*/
+    for (auto i_scalar = 0u; i_scalar < nVar; ++i_scalar) {
+      nodes->SetDiffusivity(i_point, fluid_model_local->GetMassDiffusivity(i_scalar), i_scalar);
+    }
+
+    if (!Output) LinSysRes.SetBlock_Zero(i_point);
+  }
+
+  if (config->GetComm_Level() == COMM_FULL) {
+    SU2_MPI::Reduce(&n_not_in_domain, &global_table_misses, 1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE,
+                    SU2_MPI::GetComm());
+    if (rank == MASTER_NODE) {
+      SetNTableMisses(global_table_misses);
```

Comment:
Set the member variable directly if it is only used inside the solver, not point adding more interface to the solver.

Reply:
removed this variable completely.